src/wp-admin/includes/image.php

```diff
@@ -291,30 +291,55 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 		 * If the original image's dimensions are over the threshold,
 		 * scale the image and use it as the "full" size.
 		 */
+		$scale_down = false;
+		$convert    = false;
+
 		if ( $threshold && ( $image_meta['width'] > $threshold || $image_meta['height'] > $threshold ) ) {
+			// The image will be converted if needed on saving.
+			$scale_down = true;
+		} else {
+			// The image may need to be converted regardless of its dimensions.
+			$output_format = wp_get_image_editor_output_format( $file, $imagesize['mime'] );
+
+			if ( is_array( $output_format ) && array_key_exists( $imagesize['mime'], $output_format ) ) {
+				$convert = true;
+			}
+		}
+
+		if ( $scale_down || $convert ) {
 			$editor = wp_get_image_editor( $file );
 
 			if ( is_wp_error( $editor ) ) {
 				// This image cannot be edited.
 				return $image_meta;
 			}
 
-			// Resize the image.
-			$resized = $editor->resize( $threshold, $threshold );
+			if ( $scale_down ) {
+				// Resize the image. This will also convet it if needed.
+				$resized = $editor->resize( $threshold, $threshold );
+			} elseif ( $convert ) {
+				// The image will be converted (if possible) when saved.
+				$resized = true;
+			}
```

Comment:
```suggestion
			if ( true === $scale_down ) {
				// Resize the image. This will also convet it if needed.
				$resized = $editor->resize( $threshold, $threshold );
			} elseif ( true === $convert ) {
				// The image will be converted (if possible) when saved.
				$resized = true;
			}
```

Strict comparison as mentioned above.

Reply:
See above.